src/store/period.store.ts

```diff
@@ -30,25 +30,25 @@ export class PeriodStore {
       .resize(this.period)
       .toMilliseconds()
 
-    if (!this.periods.length || this.periods[0].time < bucket) return this.newPeriod(bucket, trade)
+    if (!this.periods.length || this.periods[0].time < bucket) return this.newPeriod(bucket, price)
 
     this.periods[0].low = Math.min(price, this.periods[0].low)
-    this.periods[0].high = Math.min(price, this.periods[0].high)
+    this.periods[0].high = Math.max(price, this.periods[0].high)
     this.periods[0].close = price
     this.periods[0].volume += size
 
     // this.periodEvents.emit('newTrade')
   }
 
   @action
-  newPeriod(bucket: number, { time, size, price }: TradeItem) {
+  newPeriod(bucket: number, price: number) {
     this.periods.unshift({
       time: bucket,
       open: price,
       high: price,
       low: price,
       close: price,
-      volume: size,
+      volume: 0,
```

Comment:
This should be `size`. Since it's a new trade that came in that's outside the period of the previous period. Otherwise, the first trade from a period won't get its size recorded.